modules/bitgo/test/v2/unit/internal/tssUtils/ecdsa.ts

```diff
@@ -804,6 +816,300 @@ describe('TSS Ecdsa Utils:', async function () {
+      const adminSignedBitGoChallenge = ecdsaStaticUtils.signChallenge(rawBitGoChallenge, adminEcdhKey.xprv, derivationPath);
+      const mockChallengesResponse = {
+        enterpriseChallenge: {
+          ...rawEntChallenge,
+          verifiers: {
+            adminSignature: adminSignedEntChallenge.toString('hex'),
+          },
+        },
+        bitgoChallenge: {
+          ...rawBitGoChallenge,
+          verifiers: {
+            adminSignature: adminSignedBitGoChallenge.toString('hex'),
+          },
+        },
+        createdBy: 'id',
+      };
+      await nockGetChallenges({ walletId: walletId, response: mockChallengesResponse });
+      await ecdsaStaticUtils.getChallengesForEcdsaSigning(bitgo, walletId, entId).should.be.rejectedWith('Admin signature for enterprise challenge is not valid. Please contact your enterprise admin.');
+    });
+
+    it('Fails if the bitgo challenge signature is different from the admin ecdh key', async function() {
+      const adminSignedEntChallenge = ecdsaStaticUtils.signChallenge(rawEntChallenge, adminEcdhKey.xprv, derivationPath);
+      // Bad sign
+      const adminSignedBitGoChallenge = ecdsaStaticUtils.signChallenge(rawBitGoChallenge, fakeAdminEcdhKey.xprv, derivationPath);
+      const mockChallengesResponse = {
+        enterpriseChallenge: {
+          ...rawEntChallenge,
+          verifiers: {
+            adminSignature: adminSignedEntChallenge.toString('hex'),
+          },
+        },
+        bitgoChallenge: {
+          ...rawBitGoChallenge,
+          verifiers: {
+            adminSignature: adminSignedBitGoChallenge.toString('hex'),
+          },
+        },
+        createdBy: 'id',
+      };
+      await nockGetChallenges({ walletId: walletId, response: mockChallengesResponse });
+      await ecdsaStaticUtils.getChallengesForEcdsaSigning(bitgo, walletId, entId).should.be.rejectedWith('Admin signature for BitGo\'s challenge is not valid. Please contact your enterprise admin.');
+    });
+  });
+
+  describe('getVerifyAndSignBitGoChallenges', function() {
+    const bitgo = TestBitGo.decorate(BitGo, { env: 'mock' });
+    const adminEcdhKey = bitgo.keychains().create();
+    const derivationPath = 'm/0/0';
+    const bitgoInstChallenge = mockChallengeA;
+    const bitgoNitroChallenge = mockChallengeB;
+    const userPassword = 'password123';
+    const encryptedXprv = bitgo.encrypt({
+      password: userPassword,
+      input: adminEcdhKey.xprv,
+    });
+
+    beforeEach(async function() {
+      sinon.stub(bitgo, 'getSigningKeyForUser').resolves({
+        userId: 'id',
+        userEmail: 'user@bitgo.com',
+        derivedPubkey: bip32.fromBase58(adminEcdhKey.xpub).derivePath(derivationPath).publicKey.toString('hex'),
+        derivationPath: derivationPath,
+        ecdhKeychain: 'my keychain',
+      });
+
+      sinon.stub(bitgo, 'getECDHKeychain').resolves({
+        encryptedXprv: encryptedXprv,
+      });
```

Comment:
When you have mutliple tests like this, dont use the global sinon, use the sandbox instead.